nixos/tests/test-driver: cleanup "dead" code

blaggacao opened the pull request:
There is no real existing code path that runs 
the `create_startcommand` method.

tfc commented:
the installer tests are at least one source of tests that do use this code path.

lukegb commented:
@ofborg test installer boot

Commit:
nixos/tests/test-driver: cleanup "dead" code (USE_SERIAL)

At nixpkgs root:

`rg redirectSerial ./` does not result in any other match
nor does
`rg USE_SERIAL ./` except for an unrelated match in:

pkgs/tools/graphics/argyllcms/default.nix

blaggacao commented:
Ping: I added a second commit which apparently is dead code, too.

lukegb commented:
@ofborg test installer boot

blaggacao commented:
- Those are the tests that use `create_machine` without also providing a `startCommand`:

```console
nixos/tests/installer.nix
99:      def create_machine_named(name):
100:          return create_machine({**default_flags, "name": name})
138:      machine = create_machine_named("boot-after-install")
196:      machine = create_machine_named("rebuild-switch")
216:      machine = create_machine_named("boot-after-rebuild-switch")
226:      machine = create_machine_named("clone-default-config")
246:      machine = create_machine_named("clone-alternate-config")
```

```console
$ rg -C3 'create_machine' nixos/tests/hibernate.nix
68-  testScript =
69-    ''
70-      def create_named_machine(name):
71:          return create_machine(
72-              {
73-                  "qemuFlags": "-cpu max ${
74-                    if system == "x86_64-linux" then "-m 1024"
```

```console
$ rg -C3 'create_machine' nixos/tests/boot.nix
29-        nodes = { };
30-        testScript =
31-          ''
32:            machine = create_machine(${machineConfig})
33-            machine.start()
34-            machine.wait_for_unit("multi-user.target")
35-            machine.succeed("nix verify -r --no-trust /run/current-system")
--
69-        name = "boot-netboot-" + name;
70-        nodes = { };
71-        testScript = ''
72:            machine = create_machine(${machineConfig})
73-            machine.start()
74-            machine.wait_for_unit("multi-user.target")
75-            machine.shutdown()
```

What should we do about them? Those seem legacy ways, whereas most of `nixpkgs` seems to make use of some sort of testing instrumentation or nix-native ways of configuring the start scripts.

We could try to reduce the complexity of the testing driver subsystem and should not leave thing leaking into it's API.

blaggacao commented:
To resolve this, I'll remove the first commit and rather will implement a convenience `StartCommand` a little later so that we continue to provide a quick way to start machines without depending on any nix-native testing implementation. On the long run, I doubt this is what we want, but refactoring this here, would get me too much off track.
